pkg/registry/core/service/rest.go

```diff
@@ -81,9 +82,13 @@ func NewStorage(registry Registry, endpoints endpoint.Registry, serviceIPs ipall
 		serviceNodePorts: serviceNodePorts,
 		proxyTransport:   proxyTransport,
 	}
+	redirectOnly := true
+	if utilfeature.DefaultFeatureGate.Enabled(features.ServiceProxyAllowExternalIPs) {
+		redirectOnly = false
```

Comment:
I'm not sure we can assume everything talking to this endpoint will follow redirects